src/ui/operators/default_names.py

```diff
@@ -0,0 +1,60 @@
+"""Assign default object name as props name and object render job name."""
+from typing import Set
+
+import bpy
+
+from .... import utils
+from ....config import HANA3D_DESCRIPTION, HANA3D_NAME, HANA3D_UI
+
+
+class DefaultNamesOperator(bpy.types.Operator):
+    """Assign default object name as props name and object render job name."""
+
+    bl_idname = f'view3d.{HANA3D_NAME}_default_name'
+    bl_label = f'{HANA3D_DESCRIPTION} Default Name'
+    bl_options = {'REGISTER', 'UNDO', 'INTERNAL'}
+
+    def modal(self, context: bpy.types.Context, event: bpy.types.Event) -> Set[str]:  # noqa: D102, WPS210, E501
+        # This is for case of closing the area or changing type:
+        ui_props = getattr(context.window_manager, HANA3D_UI)
+
+        if ui_props.turn_off:
+            return {'CANCELLED'}
+
+        if event.type not in {'LEFTMOUSE', 'RIGHTMOUSE', 'ENTER'}:
+            return {'PASS_THROUGH'}
+
+        if ui_props.down_up == 'SEARCH':
+            search_props = utils.get_search_props()
+            if search_props.workspace != '' and search_props.tags_list:
+                search_props.workspace = search_props.workspace
```

Comment:
Isso serve para alguma coisa? Não parece fazer nada

Reply:
É basicamente um trick para forçar a atualizar a UI

Reply:
O ideal é conseguir atualizar de alguma outra forma, mas precisaria de uns refatoramentos maiores.

Reply:
Entendi. Vou colocar um comentário explicando isso (com um TODO para resolver um dia)